include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -962,6 +962,164 @@ class open_addressing_ref_impl {
     }
   }
 
+  /**
+   * @brief Executes a callback on every element in the container with key equivalent to the probe
+   * key.
+   *
+   * @note Passes an un-incrementable input iterator to the element whose key is equivalent to
+   * `key` to the callback.
+   *
+   * @tparam ProbeKey Input type which is convertible to 'key_type'
+   * @tparam CallbackOp Unary callback functor or device lambda
+   *
+   * @param key The key to search for
+   * @param callback_op Function to call on every element found
+   */
+  template <class ProbeKey, class CallbackOp>
+  __device__ void for_each(ProbeKey const& key, CallbackOp&& callback_op) const noexcept
```

Comment:
```suggestion
  __device__ void for_each(ProbeKey const& key, CallbackOp&& callback_op) const
```
we need extra attention to deal with `noexcept` in the future since this function may throw.

Reply:
nonsense here. sorry about the confusion.